Python/codecs.c

```diff
@@ -127,8 +127,7 @@ PyObject *_PyCodec_Lookup(const char *encoding)
         return result;
     }
     else if (PyErr_Occurred()) {
-        Py_DECREF(v);
-        return NULL;
+        goto onError;
     }
```

Comment:
I would prefer to clear v reference just after the PyDict_GetItemWithError() call. There is no need to keep alive.

I also suggest to move "v" variable declaration to where it's defined ("v = ...").

Reply:
> I would prefer to clear v reference just after the PyDict_GetItemWithError() call.

That doesn't work, the string `v` is used in multiple places. We really need to keep it alive until the end.

Reply:
> That doesn't work, the string v is used in multiple places. We really need to keep it alive until the end.

Oh ok, I didn't notice.